packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_model.dart

```diff
@@ -3,37 +3,180 @@
 // found in the LICENSE file.
 
 /// Contains all data relevant to a deep link.
-class LinkData {
+///
+import 'package:flutter/material.dart';
+
+import '../../shared/primitives/utils.dart';
+import '../../shared/table/table_data.dart';
+import '../../shared/theme.dart';
+import '../../shared/ui/search.dart';
+import '../../shared/utils.dart';
+
+const kDeeplinkTableCellDefaultWidth = 200.0;
+
+class LinkData with SearchableDataMixin {
   LinkData({
-    required this.os,
     required this.domain,
-    required this.paths,
-    this.scheme = 'Http://, Https://',
+    required this.path,
+    required this.os,
+    this.scheme = const <String>['Http://', 'Https://'],
     this.domainError = false,
     this.pathError = false,
   });
 
-  final String os;
-  final List<String> paths;
+  final String path;
   final String domain;
-  final String scheme;
+  final List<String> os;
+  final List<String> scheme;
   final bool domainError;
   final bool pathError;
 
-  String get searchLabel => (os + paths.join() + domain + scheme).toLowerCase();
+  @override
+  bool matchesSearchToken(RegExp regExpSearch) {
+    return (domain.caseInsensitiveContains(regExpSearch) == true) ||
+        (path.caseInsensitiveContains(regExpSearch) == true);
+  }
 
-  LinkData mergeByDomain(LinkData? linkData) {
-    if (linkData == null) {
-      return this;
-    }
+  @override
+  String toString() => 'LinkData($domain $path)';
+}
+
+class DomainColumn extends ColumnData<LinkData> {
+  DomainColumn()
+      : super(
+          'Domain',
+          fixedWidthPx: scaleByFontFactor(kDeeplinkTableCellDefaultWidth),
+        );
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  String getValue(LinkData dataObject) => dataObject.domain;
+
+  @override
+  Widget getDisplayWidget(LinkData dataObject, BuildContext context) {
+    return Row(
+      children: [
+        if (dataObject.domainError)
+          Padding(
+            padding: const EdgeInsets.only(right: denseSpacing),
+            child: Icon(
+              Icons.error,
+              color: Theme.of(context).colorScheme.error,
+            ),
+          ),
```

Comment:
consider adding an `else` statement that adds a `SizedBox()` of the same width as the total Padding + Icon width, so that the table rows do not have staggered indents depending on whether there is an error or not

Reply:
In our design mock, icons are aligned together with text like 
<img width="471" alt="5D8DC136-CBC2-4EAC-A327-7C366B0A7817" src="https://github.com/flutter/devtools/assets/108393416/6e867f5c-cdf6-44fd-8732-01fc7b2ff104">